pulser/sequence.py

```diff
@@ -133,39 +144,46 @@ def __init__(self, register, device):
         # Checks if register is compatible with the device
         device.validate_register(register)
 
-        self._register = register
-        self._device = device
-        self._in_xy = False
-        self._calls = [_Call("__init__", (register, device), {})]
-        self._channels = {}
-        self._schedule = {}
-        self._phase_ref = {}  # The phase reference of each channel
-        # Stores the names and corresponding ids of declared channels
-        self._taken_channels = {}
-        self._qids = set(self.qubit_info.keys())  # IDs of all qubits in device
-        self._last_used = {}    # Last time each qubit was used, by basis
-        self._last_target = {}  # Last time a target happened, by channel
+        self._register: Register = register
+        self._device: Device = device
+        self._in_xy: bool = False
+        self._calls: list[_Call] = [_Call("__init__", (register, device), {})]
+        self._channels: dict[str, Channel] = {}
+        self._schedule: dict[str, list[_TimeSlot]] = {}
+        # The phase reference of each channel
+        self._phase_ref: dict[str, dict[QubitId, _PhaseTracker]] = {}
+        # Stores the names and dict ids of declared channels
+        self._taken_channels: dict[str, str] = {}
+        # IDs of all qubits in device
+        self._qids = set(self.qubit_info.keys())
+        # Last time each qubit was used, by basis
+        self._last_used: dict[str, dict[QubitId, int]] = {}
+        # Last time a target happened, by channel
+        self._last_target: dict[str, int] = {}
+        self._variables: dict[str, Variable] = {}
+        self._to_build_calls: list[_Call] = []
+        self._building: bool = True
 
         # Initializes all parametrized Sequence related attributes
         self._reset_parametrized()
 
     @property
-    def qubit_info(self):
+    def qubit_info(self) -> dict[QubitId, ArrayLike]:
```

Comment:
If you change this to:
```suggestion
    def qubit_info(self) -> dict[QubitId, np.ndarray]:
```
it should fix the mypy error.